### PR TITLE
AC Sound Searcher

### PR DESCRIPTION
**Pull request by ericrius1**
This PR provides a way of creating sounds that are injected from an AC script. The AC script listens for messages from entities with a special userData field that provides information about the sound to be played (url, what kind of interval to play at, volume, etc). Also contained in this PR is an example of a client script which spawns an entity of this type

### CODE REVIEW

**Comment by ericrius1** [comment]
To Test: 
1) run https://rawgit.com/ericrius1/hifi/acSoundSearcher/examples/audioExamples/acAudioSearching/ACAudioSearchAndInject.js as an assignment script

2) Run https://rawgit.com/ericrius1/hifi/acSoundSearcher/examples/audioExamples/acAudioSearching/acAudioSearchCompatibleEntitySpawner.js from the client. Make sure you are at position (0, 0, 0)

3) You should hear a dove singing its haunting song at every 2-6 seconds

Look at lines 22-27 of acAudioSearchCompatibleEntitySpawner.js for the correct format of the userData field specifying the sound data.

From https://app.asana.com/0/31759584831097/86123728670216

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89d235e2c2a9dc08901cb77626a22782fbee1da2 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a95176be8fc25a58e0d272119a1efc874b110cc3 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by nbq** [comment]
Testing this now

**Comment by nbq** [comment]
I have followed the instructions on two domains now, setting the AC script and then the client script, the client script creates an object but no sound ever plays. I assume there may be an issue with the AC script.

**Comment by RyanDowne** [comment]
I also tested this but it didn't seem to work. I am using the client from this PR, I am running the AC script on the HQ server and I see the agent in the node list, I created the entity at 0,0,0 and see it there with the user data filled out and the script attached, I checked the wav url and it does play in a browser.

**Comment by huffman** [comment]
Maybe I am misunderstanding the goal of this, but why use Messages at all? If you use the `Entities.[adding|deleting]Entity(entityID)` events you can inspect the `userData` of entities as they come in.

**Comment by ZappoMan** [comment]
I'm with @huffman on this one... DON'T use messages in this script... it makes no sense, is not needed, and is only going to complicate your script. We need to do a design meeting about why you guys are so insistent on using messages... they RARELY are needed or useful.

File: examples/audioExamples/acAudioSearching/soundEntityScript.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.MESSAGE_CHANNEL = "Hifi-Sound-Entity";
+    };
+
+    SoundEntity.prototype = {
+        preload: function(entityID) {
+            this.entityID = entityID;
+            Script.setTimeout(function() {
+                var soundData = getEntityCustomData("soundKey", _this.entityID);
+                // Just a hack for now until bug https://app.asana.com/0/26225263936266/86767805352289 is fixed
+                Messages.sendMessage(_this.MESSAGE_CHANNEL, JSON.stringify({
```
Comment (ZappoMan) [inline examples/audioExamples/acAudioSearching/soundEntityScript.js]: remove this code... this makes no sense. Don't use messages like this.

**Comment by ZappoMan** [comment]
this design is not at all what I believed we discussed... the entitys should not need scripts at all... all of the work should be done in the AC.

**Comment by ericrius1** [comment]
@huffman- just discussed with @ZappoMan, and adding and deletingEntity events do not work from within an AC script.

@nbq and @RyanDowne - thanks for testing. I'm going to change the design, so will be a waste to figure out why this iteration isn't working for you. I'll update when feature is ready to be tested again

**Comment by huffman** [comment]
@ericrius1 I am skeptical of that, I use them myself.

**Comment by ericrius1** [comment]
@huffman- Awesome! I'll give it a shot. That'd make things simple

**Comment by huffman** [comment]
@ericrius1 np, if you need help let me know, I should be able to dig up a few examples.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 03f90c20706f1c54e5ddff1d6c2baf9a507f5506 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3fe20c818151163bcb3b9559eed32604cf67d85 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8591de4873d5b16eeed74b79dd7fed64536e860b available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c8704234b19dd866fe69ed3469fdd991a4a66f96 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 16af9b9e3165869e6717c879cb45cf6ae0c26d3a available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b0e20d82b2236ff21af6aeeff05038847e8af22 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa49db9992f04b9eecd93e1c561f84d5c5acef1e available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 355f60f0888c0dd244946ce86ef0e1d31a7b5ca5 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 609ddda74e91b47d024a24f72a5c4ed5d5b06eeb available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 609ddda74e91b47d024a24f72a5c4ed5d5b06eeb available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 208007c9d88dcbf8b8e7f6b666db13fec0319611 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a5c037b23fcedfcc8d737a0a8e79d4a07e149ca0 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1c54de6f571b2d67201cfd78c2341de849e630d9 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4c33c6e1710400738eb5123d87455cfc1a397ad0 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d64da2df73fb7e40d3f953ff0fe5d188f24562a available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 053f163b9276718d8c60af8185f805e637617968 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0932df621a62a5944d352af57e2c1b73d581e140 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ab3bd204ddfc7007a1aef0fbdca0fdea285fa60 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 193857f3449eba4f8cad1dab072bd149eb0efd5e available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b240dfd5f14fbbb7848bf308bbb239ae23d5025 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b240dfd5f14fbbb7848bf308bbb239ae23d5025 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c169a7cdd5e201b86f946304aa9a4eff51988cd9 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bad4a5d6019565c69b253954ac936d47f37587b6 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by ericrius1** [comment]
This is QA ready- please run this build since some changes were made to c++

**Comment by hifi-gustavo** [comment]
Build console output for commit: b60ba0c5a7d694f9983b7bc7b544ac7aef74ad6c available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 85ddb7100c9bfd6d23b82aab8f3714ffe0114d52 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3acc077bbd45f4ec637c54ea316d115e2ed86cb9 available at: https://gist.github.com/c2bb01d838697ed10ccb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7023.dmg

**Pull request merged**